app/web/controllers/audio_files.py

```diff
@@ -14,6 +17,37 @@
 logger = logging.getLogger(__name__)
 
 
+@dataclass
+class DirEntry:
+    name: str
+    prefix: str
+    file_callback: Optional[str]
+
+    @property
+    def url(self) -> str:
+        dir_prefix = f'{self.prefix}/{self.name}' if self.prefix else self.name
+        query = to_query_string({'prefix': dir_prefix, 'file_callback': self.file_callback})
+        return f"/audio_files?{query}"
+
+
+@dataclass
+class FileEntry:
+    name: str
+    prefix: str
+    file_callback: Optional[str]
+
+    @property
+    def url(self) -> Optional[str]:
+        if self.file_callback is not None:
+            file_opt = self.name if self.prefix == '' else f"{self.prefix}/{self.name}"
```

Comment:
to be the same as line 28

```suggestion
            file_opt = f'{self.prefix}/{self.name}' if self.prefix else self.name
```